pl_examples/quantization/imagenet.py

```diff
@@ -0,0 +1,374 @@
+        outputs = self.validation_epoch_end(*args, **kwargs)
+
+        def substitute_val_keys(out):
+            return {k.replace("val", "test"): v for k, v in out.items()}
+
+        outputs = {
+            "test_loss": outputs["val_loss"],
+            "progress_bar": substitute_val_keys(outputs["progress_bar"]),
+            "log": substitute_val_keys(outputs["log"]),
+        }
+        return outputs
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):  # pragma: no-cover
+        parser = parent_parser.add_argument_group("ImageNetLightningModel")
+        parser.add_argument(
+            "-a",
+            "--arch",
+            metavar="ARCH",
+            default="resnet18",
+            choices=ImageNetLightningModel.MODEL_NAMES,
+            help=("model architecture: " + " | ".join(ImageNetLightningModel.MODEL_NAMES) + " (default: resnet18)"),
+        )
+        parser.add_argument(
+            "-j", "--workers", default=4, type=int, metavar="N", help="number of data loading workers (default: 4)"
+        )
+        parser.add_argument(
+            "-b",
+            "--batch-size",
+            default=256,
+            type=int,
+            metavar="N",
+            help="mini-batch size (default: 256), this is the total batch size of all GPUs on the current node"
+            " when using Data Parallel or Distributed Data Parallel",
+        )
+        parser.add_argument(
+            "--lr", "--learning-rate", default=0.001, type=float, metavar="LR", help="initial learning rate", dest="lr"
+        )
+        parser.add_argument("--momentum", default=0.9, type=float, metavar="M", help="momentum")
+        parser.add_argument(
+            "--wd",
+            "--weight-decay",
+            default=1e-4,
+            type=float,
+            metavar="W",
+            help="weight decay (default: 1e-4)",
+            dest="weight_decay",
+        )
+        parser.add_argument("--pretrained", dest="pretrained", action="store_true", help="use pre-trained model")
+        return parent_parser
+
+
+def main(args: Namespace) -> None:
+    if args.seed is not None:
+        pl.seed_everything(args.seed)
+
+    print(args)
+
+    if args.accelerator == "ddp":
+        # When using a single GPU per process and per
+        # DistributedDataParallel, we need to divide the batch size
+        # ourselves based on the total number of GPUs we have
+        args.batch_size = int(args.batch_size / max(1, args.gpus))
+        args.workers = int(args.workers / max(1, args.gpus))
+
+    model = ImageNetLightningModel(**vars(args))
+    if args.evaluate:
+        if args.precision_mode == 'int8':
+            model = pl.compressor.intel.neural_compressor.load(model, args.checkpoint)
+        dataloader = model.val_dataloader()
+        validate(dataloader, model, args)
+        return
+
+    quantizer = pl.IncQuantization('config/quantization.yaml', model, calib_dataloader=model.val_dataloader(),
```

Comment:
This should be a callback. Find the [SparseML Integration from DeepSparse](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/callbacks/sparseml.py)

Reply:
Once the callback is done, let's have it along `QuantizationAwareTraining` in `pytorch_lightning/callbacks/quantization.py`

Reply:
callback is done